src/pykeen/models/unimodal/rgcn/model.py

```diff
@@ -0,0 +1,302 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of the R-GCN model."""
+
+import logging
+from os import path
+from typing import Any, Mapping, Optional
+
+import torch
+from class_resolver import Hint, Resolver
+from torch import nn
+
+from .decompositions import Decomposition, decomposition_resolver
+from .weightings import EdgeWeighting, edge_weight_resolver
+from ...nbase import ERModel, EmbeddingSpecificationHint
+from ....nn import EmbeddingSpecification, RepresentationModule
+from ....nn.modules import Interaction, interaction_resolver
+from ....triples import TriplesFactory
+from ....typing import Initializer, RelationRepresentation
+
+__all__ = [
+    "RGCN",
+]
+
+logger = logging.getLogger(name=path.basename(__file__))
+
+
+class Bias(nn.Module):
+    """A module wrapper for adding a bias."""
+
+    def __init__(self, dim: int):
+        """Initialize the module.
+
+        :param dim: >0
+            The dimension of the input.
+        """
+        super().__init__()
+        self.bias = nn.Parameter(torch.empty(dim), requires_grad=True)
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        """Reset the layer"s parameters."""
+        nn.init.zeros_(self.bias)
+
+    # pylint: disable=arguments-differ
+    def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
+        """Add the learned bias to the input.
+
+        :param x: shape: (n, d)
+            The input.
+
+        :return:
+            x + b[None, :]
+        """
+        return x + self.bias.unsqueeze(dim=0)
+
+
+# TODO: Move to utils
+activation_resolver = Resolver.from_subclasses(base=nn.Module, default=nn.ReLU)
+
+
+class RGCNRepresentations(RepresentationModule):
+    """Entity representations enriched by R-GCN."""
+
+    def __init__(
+        self,
+        triples_factory: TriplesFactory,
+        embedding_specification: EmbeddingSpecificationHint,
+        num_layers: int = 2,
+        use_bias: bool = True,
+        use_batch_norm: bool = False,
+        activation: Hint[nn.Module] = None,
+        activation_kwargs: Optional[Mapping[str, Any]] = None,
```

Comment:
@cthoyt do we have a type annotation for this?  we use it quite often.

Reply:
No we don't, but I think it's worth leaving this explicit so we never forget what it is